homeassistant/components/hive/config_flow.py

```diff
@@ -107,21 +106,17 @@ async def async_step_configuration(self, user_input=None):
         errors = {}
 
         if user_input:
-            if not errors:
-                try:
-                    if self.device_registration:
-                        await self.hive_auth.device_registration(
-                            user_input["device_name"]
-                        )
-                        self.data[
-                            "device_data"
-                        ] = await self.hive_auth.get_device_data()
-                    return await self.async_setup_hive_entry()
-                except UnknownHiveError:
-                    errors["base"] = "unknown"
+            try:
+                if self.device_registration:
+                    self.device_name = user_input["device_name"]
+                    await self.hive_auth.device_registration(user_input["device_name"])
+                    self.data["device_data"] = await self.hive_auth.get_device_data()
```

Comment:
Can these calls raise UnknownHiveError? If not, please move them out of the try... except block.